Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformProjectsController.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.domain.network.INetwork;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.dto.TransformFilesDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.IEditorManager;
+import edu.asu.spring.quadriga.service.IUserManager;
+import edu.asu.spring.quadriga.service.network.INetworkManager;
+import edu.asu.spring.quadriga.service.transformation.ITransformationManager;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+
+/**
+ * This class will list all the networks approved by the user and display
+ * approved networks and some dummy transformations for the data
+ * 
+ * @author: Jaydatta Nagarkar.
+ * 
+ */
+@Controller
+public class TransformProjectsController {
+    @Autowired
+    private ITransformationManager transformManager;
+
+    
+    @Autowired
+    private INetworkManager networkManager;
+    
+    @Autowired
+    private IRetrieveProjectManager retrieveProjectManager;
+
+    @Autowired
+    private IEditorManager editorManager;
+
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IProject projectManager;
+
+    private static final Logger logger = LoggerFactory
+            .getLogger(TransformProjectsController.class);
+
+    /**
+     * List networks assigned to a User
+     * 
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     */
+    @RequestMapping(value = "auth/transformation", method = RequestMethod.GET)
+    private String listTransformations(ModelMap model, Principal principal)
+            throws QuadrigaStorageException {
+        IUser user = userManager.getUser(principal.getName());
+        Set<IProject> projects = new HashSet<>();
+        Map<String, List<INetwork>> networkMap = new HashMap<>();
+        List<IProject> retrievedProjects=new ArrayList<IProject>();
+        try {
+            retrievedProjects=retrieveProjectManager.getProjectList(user.getName());        
+        } catch (QuadrigaStorageException e) {
+            logger.error("Error fetching list of approved networks", e);
+            model.addAttribute("show_error_alert", true);
+            model.addAttribute("error_alert_msg",
+                    "There was an error retrieving the list of approved networks.");
+            return "auth/transformation";
+        }
+        List<INetwork> networks = new ArrayList<INetwork>();
+        for(int i=0; i < retrievedProjects.size();i++){
```

Comment:
hm, actually not sure why that throws a null pointer right now (it shouldn't), but in any case, you do need to check.